core/corehttp/gateway_handler.go

```diff
@@ -417,20 +416,23 @@ func (i *gatewayHandler) putHandler(w http.ResponseWriter, r *http.Request) {
 	ctx, cancel := context.WithCancel(i.node.Context())
 	defer cancel()
 
-	rootPath, err := path.ParsePath(r.URL.Path)
+	rootPath, err := coreapi.ParsePath(r.URL.Path)
 	if err != nil {
 		webError(w, "putHandler: IPFS path not valid", err, http.StatusBadRequest)
 		return
 	}
 
-	rsegs := rootPath.Segments()
+	rsegs := rootPath.Components()
 	if rsegs[0] == ipnsPathPrefix {
 		webError(w, "putHandler: updating named entries not supported", errors.New("WritableGateway: ipns put not supported"), http.StatusBadRequest)
 		return
 	}
 
 	var newnode node.Node
 	if rsegs[len(rsegs)-1] == "QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn" {
+		// TODO(lgierth): this seems to mean that PUT /ipfs/QmUNLL/foo/bar
+		//                results in bar being an empty directory,
+		//                no matter what
 		newnode = ft.EmptyDirNode()
```

Comment:
It seems that `QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn` is the hash for an empty directory.